### ISSUE TITLE
seed for DistributedSampler

### ISSUE THREAD

**Issue by leezu** [PyTorchLightning/pytorch-lightning#7016]
## 🐛 Bug
[torch.utils.data.distributed.DistributedSampler](https://pytorch.org/docs/stable/data.html#torch.utils.data.distributed.DistributedSampler) defaults to random seed `0` to shuffle the sampler if `shuffle=True`.
As lightning does not specify a seed, multi-gpu training with DistributedSampler in lightning will use a deterministic order independently from the user-specified seed (`seed_everything`).

https://github.com/PyTorchLightning/pytorch-lightning/blob/b85cfbe8f350a89c93ff967fa416859be7ebb4f3/pytorch_lightning/trainer/data_loading.py#L198-L202

https://github.com/PyTorchLightning/pytorch-lightning/blob/33cc9fe13838b3f97d0ee9a573900760aa6b63f4/pytorch_lightning/plugins/training_type/ddp.py#L91-L94

As a workaround, users can specify `replace_sampler_ddp` when creating the Trainer, but I think this is something that lightning should handle well out of the box.

**Comment by awaelchli** [comment]
let's add it :) 
we can read the seed from the `PL_GLOBAL_SEED` env variable :)

**Comment by leezu** [comment]
Great, thanks for pointing out the variable and opening the PR :)!

**Comment by awaelchli** [comment]
I will remove the bug label and treat it as an enhancement, because the default DistributedSampler sets a default value for the seed, so experiments are still reproducible. 
However, note that if we make that change, we cannot guarantee full parity from one version to the other.
